pkg/controllermanager/controller/cloudprofile/cloudprofile_control.go

```diff
@@ -91,11 +94,13 @@ type ControlInterface interface {
 
 // NewDefaultControl returns a new instance of the default implementation ControlInterface that
 // implements the documented semantics for CloudProfiles.
-func NewDefaultControl(k8sGardenClient kubernetes.Interface, shootLister gardencorelisters.ShootLister, recorder record.EventRecorder) ControlInterface {
-	return &defaultControl{k8sGardenClient, shootLister, recorder}
+func NewDefaultControl(ctx context.Context, config *controllermgrconfig.CloudProfileControllerConfiguration, k8sGardenClient kubernetes.Interface, shootLister gardencorelisters.ShootLister, recorder record.EventRecorder) ControlInterface {
+	return &defaultControl{ctx, config, k8sGardenClient, shootLister, recorder}
 }
 
 type defaultControl struct {
+	ctx             context.Context
```

Comment:
I guess it is not a good practice to store context in a struct type.

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

Ref https://golang.org/pkg/context/